user_sync/helper.py

```diff
@@ -111,12 +111,11 @@ def read_csv_rows(cls, file_path, recognized_column_names=None, logger=None, enc
                     if len(unrecognized_column_names) > 0 and logger is not None:
                         logger.warn("In file '%s': unrecognized column names: %s", file_path, unrecognized_column_names)
                 for row in reader:
+                    row.pop(None, None)
                     if is_py2():
-                        # in py2, we need to decode both the column names *and* the values
-                        newrow = {}
                         for key, val in six.iteritems(row):
-                            newrow[key.decode(encoding, 'strict')] = val.decode(encoding, 'strict') if val else None
-                        row = newrow
+                            row[key.decode(encoding, 'strict')] = val.decode(encoding, 'strict') if val else val
```

Comment:
It generally isn't safe to mutate a data structure returned from an iterable in a loop.  I recommend mutating `newrow` and also yielding `newrow`.

```python
newrow[key.decode(encoding, 'strict')] = val.decode(encoding, 'strict') if val else val
# ... (outside of row loop)
yield newrow
```

Reply:
Sure thing - I'll revert that as it was initially